### PR TITLE
CC-31568: Extended the warehouse picking guide with the FA OAuth.

### PR DESCRIPTION
**Pull request by abitskil**
* Epic: https://spryker.atlassian.net/browse/CC-31691

* Ticket: https://spryker.atlassian.net/browse/CC-31568

### CODE REVIEW

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 1]
```diff
@@ -108,6 +132,24 @@ Make sure the following changes have been triggered in transfer objects:
...
+| spy_oauth_code_flow_auth_code | table | created |
+
+
+Make sure the following changes have been triggered in transfer objects:
+
+| TRANSFER                   | TYPE  | EVENT   | PATH                                                             |
+|----------------------------|-------|---------|------------------------------------------------------------------|
+| AuthCode                   | class | created | src/Generated/Shared/Transfer/AuthCodeTransfer                   |
+| AuthCodeAttributes         | class | created | src/Generated/Shared/Transfer/AuthCodeAttributesTransfer         |
+| AuthCodeResponseAttributes | class | created | src/Generated/Shared/Transfer/AuthCodeResponseAttributesTransfer |
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:151]: Minor: there are a lot of transfer properties existing in OauthCodeFlow only as OauthRequest.code, OauthRequest.codeChallengeMethod, OauthRequest.codeVerifier, OauthRequest.codeChallenge, OauthRequest.state, OauthResponse.state, OauthResponse.code, etc. Please add only if you think it makes sense.

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 2]
```diff
@@ -287,6 +329,44 @@ class PickingListPushNotificationConfig extends SprykerPickingListPushNotificati
...
+    "secret": "frontend-secret",
+    "isConfidential": true,
+    "name": "Name of your app",
+    "redirectUri": "https://my-app.com",
+    "isDefault": true
+  }
+]
+```
+
+For security reasons, it is recommended to use a different OAUTH client for each application.
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:349]: Lets be consistent and use or `OAuth` or `OAUTH`

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 3]
```diff
@@ -287,6 +329,44 @@ class PickingListPushNotificationConfig extends SprykerPickingListPushNotificati
...
+
+{% info_block warningBox "Verification" %}
+
+Ensure that the OAuth client exists in the database:
+
+* Run the following SQL query:
+```sql
+SELECT * FROM spy_oauth_client WHERE identifier = 'the-client-identifier-of-your-app';
+```
+* Check that the output contains one record (create if not) and the redirect uri is not empty.
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:366]:
```suggestion
* Check that the output contains one record (create if not) and the redirect URI is not empty.
```

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 4]
```diff
@@ -550,7 +630,117 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
 }
 ```
 
-3. To enable the Backend API, register these plugins:
+3. If you've installed the Fulfillment Apps's OAuth Authorization module in [Install the required modules](#install-the-required-modules), enable the following plugins:
+
+| PLUGIN                                                       | SPECIFICATION                                                                                      | PREREQUISITES | NAMESPACE                                                                                  |
+|--------------------------------------------------------------|----------------------------------------------------------------------------------------------------|---------------|--------------------------------------------------------------------------------------------|
+| AuthorizeResource                                            | Registers the OAuth authorize resource.                                                            |               | SprykerEco\Glue\AuthorizationPickingAppBackendApi\Plugin\GlueApplication\AuthorizeResource |
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:637]:
```suggestion
| AuthorizeResource                                            | Registers the OAuth authorize resource.                                                            |               | SprykerEco\Glue\AuthorizationPickingAppBackendApi\Plugin\GlueApplication|
```

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 5]
```diff
@@ -550,7 +630,117 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
...
+
+**src/Pyz/Glue/GlueBackendApiApplication/GlueBackendApiApplicationDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Glue\GlueBackendApiApplication;
+
+use Spryker\Glue\GlueBackendApiApplication\GlueBackendApiApplicationDependencyProvider as SprykerGlueBackendApiApplicationDependencyProvider;
+use SprykerEco\Glue\AuthorizationPickingAppBackendApi\Plugin\GlueApplication\AuthorizeResource;
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:648]:
```suggestion
use SprykerEco\Glue\AuthorizationPickingAppBackendApi\Plugin\GlueApplication;
```

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 6]
```diff
@@ -108,6 +132,46 @@ Make sure the following changes have been triggered in transfer objects:
...
+
+
+Make sure the following changes have been triggered in transfer objects:
+
+| TRANSFER                               | TYPE     | EVENT   | PATH                                                     |
+|----------------------------------------|----------|---------|----------------------------------------------------------|
+| AuthCode.code                          | property | created | src/Generated/Shared/Transfer/AuthCodeTransfer           |
+| OauthRequest.responseType              | property | created | src/Generated/Shared/Transfer/OauthRequestTransfer       |
+| OauthRequest.redirectUri               | property | created | src/Generated/Shared/Transfer/OauthRequestTransfer       |
+| OauthRequest.responseType              | property | created | src/Generated/Shared/Transfer/OauthRequestTransfer       |
```
Comment (ilyakubanov) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:152]: duplicate

**Review by ilyakubanov — DISMISSED** [review]

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 7]
```diff
@@ -108,6 +132,46 @@ Make sure the following changes have been triggered in transfer objects:
...
+| DATABASE ENTITY               | TYPE  | EVENT   |
+|-------------------------------|-------|---------|
+| spy_oauth_code_flow_auth_code | table | created |
+
+
+Make sure the following changes have been triggered in transfer objects:
+
+| TRANSFER                               | TYPE     | EVENT   | PATH                                                     |
+|----------------------------------------|----------|---------|----------------------------------------------------------|
+| AuthCode.code                          | property | created | src/Generated/Shared/Transfer/AuthCodeTransfer           |
```
[the remark this answers is not in the document]
Comment (abitskil) [reply _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:149]: This module brings an additional properties also (https://github.com/spryker-eco/authorization-picking-app-backend-api),

[the remark this answers is not in the document]
Comment (ilyakubanov) [reply _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md]: There is not AuthCode.code

**Review by ilyakubanov — DISMISSED** [review]

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 8]
```diff
@@ -108,11 +134,54 @@ Make sure the following changes have been triggered in transfer objects:
...
+| AuthCodeAttributes.state               | property | created | src/Generated/Shared/Transfer/AuthCodeAttributesTransfer |
+| AuthCodeAttributes.codeChallenge       | property | created | src/Generated/Shared/Transfer/AuthCodeAttributesTransfer |
+| AuthCodeAttributes.codeChallengeMethod | property | created | src/Generated/Shared/Transfer/AuthCodeAttributesTransfer |
+| AuthCode.identifier                    | property | created | src/Generated/Shared/Transfer/AuthCodeTransfer           |
+| AuthCode.expiresAt                     | property | created | src/Generated/Shared/Transfer/AuthCodeTransfer           |
+
 {% endinfo_block %}
 
+
 ## 3) Set up configuration
```
Comment (andriitserkovnyi) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:178]: I split this section into subsections. Please make sure the subsection titles are relevant

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 9]
```diff
@@ -550,7 +660,117 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
 }
 ```
 
-3. To enable the Backend API, register these plugins:
+3. Optional: If you've installed the OAuth Authorization, enable the following plugins:
+
+| PLUGIN                                                       | SPECIFICATION                                                                                      | PREREQUISITES | NAMESPACE                                                                |
+|--------------------------------------------------------------|----------------------------------------------------------------------------------------------------|---------------|--------------------------------------------------------------------------|
+| AuthorizeResource                                            | Registers the OAuth authorize resource.                                                            |               | SprykerEco\Glue\AuthorizationPickingAppBackendApi\Plugin\GlueApplication |
+| UserAuthCodeOauthRequestGrantTypeConfigurationProviderPlugin | Builds `OauthGrantTypeConfigurationTransfer` from the configuration of AuthorizationCode GrantType data. |               | Spryker\Zed\OauthCodeFlow\Communication\Plugin\Oauth                     |
```
Comment (andriitserkovnyi) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:668]: Are `AuthorizationCode` and `GrantType` separate code references?

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 10]
```diff
@@ -550,7 +660,116 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
...
+Content-Length: 210
+
+username={username}&password={password}&response_type=code&client_id={client_id}&state={state}&code_challenge={code_challenge}&code_challenge_method=S256&redirect_uri={redirect_uri}
+```
+
+| PARAMETER  | TYPE   | EXAMPLE                           | DESCRIPTION |
+|-----------------------|--------|-----------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| username              | string | some@user.com                     | The username of the Back Office user. |
+| password              | string | 349dldks239fj93498di          | The password of the Back Office user.                        |
+| response_type         | string | code   | Defines how the authorization server should respond to the client after the resource owner grants. |
```
Comment (andriitserkovnyi) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:738]: grants what?

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 11]
```diff
@@ -550,7 +660,116 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
...
+| client_id             | string | the-client-identifier-of-your-app | Public identifier for the client application that is requesting access to a user's resources. |
```
Comment (andriitserkovnyi) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:739]: Provide proper examples. See other parameters as well

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md [hunk 12]
```diff
@@ -550,7 +662,118 @@ class PickingListDependencyProvider extends SprykerPickingListDependencyProvider
...
+|----------------|--------|-----------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| grant_type     | string | authorization_code                | Specifies the type of grant requested by the client. In this case, it's the Authorization Code Grant flow. |
+| code           | string | some-code                         | The authorization code provided by the OAuth server. |
+| client_id      | string | the-client-identifier-of-your-app | Public identifier for the client application that is requesting access to a user's resources. |
+| code_verifier  | string | some-random-string                | A random string generated by the client in the Authorization Code Grant flow with PKCE. It's used to verify the identity of the client when exchanging the authorization code for an access token. |
```
Comment (andriitserkovnyi) [inline _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:768]: Provide the actual examples

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md (hunk 9 above)
Comment (abitskil) [reply _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:672]: Only the `AuthorizationCode`.

File: _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md (hunk 10 above)
Comment (abitskil) [reply _includes/pbc/all/install-features/202311.0/unified-commerce/install-the-warehouse-picking-feature.md:740]: Extended the sentence.

**Pull request merged**